paymentsheet/src/main/java/com/stripe/android/paymentsheet/viewmodels/BaseSheetViewModel.kt

```diff
@@ -157,6 +160,11 @@ internal abstract class BaseSheetViewModel(
 
     val currentScreen: StateFlow<PaymentSheetScreen?> = backStack
         .map { it.lastOrNull() }
+        .onEach { screen ->
+            if (screen != null) {
+                reportNavigationEvent(screen)
```

Comment:
What happens on a re-subscription? It seems like these events should be reported when they happen, not when they're consumed.

Reply:
Fair point. I’ll move it to `transitionTo()`.